### PR TITLE
Added enhancer badge to player profile

### PR DESCRIPTION
**Pull request by Morabotti**
Added enhancer badge to player's profile. Thought this was nice and will prolly boost VIP sells.

Picture of the feature:
normal

Edge case (Banned user with badge):
profile-update-banned

### CODE REVIEW

File: src/content/helpers/player-profile.js [hunk 1]
```diff
@@ -4,6 +4,13 @@ import { getCurrentPath } from './location'
 export const getPlayerProfileNickname = path => {
   const match = /players(?:-modal)?\/([^/]+)\//.exec(path || getCurrentPath())
 
+  if (!match || !match[1]) {
+    const profile = /players(?:-modal)?\/([^/]+)$/.exec(
+      path || getCurrentPath()
+    )
+    return profile && profile[1]
+  }
+
```
Comment (timche) [inline src/content/helpers/player-profile.js:13]: Why is this required and how is this different from the `.exec` above?

File: src/content/index.js [hunk 2]
```diff
@@ -79,15 +80,19 @@ function observeBody() {
...
+        if (modals.isPlayerProfileStats()) {
+          runFeatureIf(
+            'playerProfileLevelProgress',
+            addPlayerProfileLevelProgress,
+            modalElement
+          )
+          addPlayerProfileDownloadDemo(modalElement)
+          addProfileMatchesEloPoints(modalElement)
+          addPlayerProfileExtendedStats(modalElement)
+        }
```
Comment (timche) [inline src/content/index.js:95]: Please keep this outside of `} else if (modals.isPlayerProfile()) {`.

File: src/content/index.js [hunk 3]
```diff
@@ -147,20 +152,29 @@ function observeBody() {
...
+            if (
+              mutation.type === 'childList' &&
+              mutation.target === statsTable
+            ) {
+              addProfileMatchesEloPoints(mainContentElement)
+            }
+          })
+          addPlayerProfileDownloadDemo(mainContentElement)
+          addPlayerProfileExtendedStats(mainContentElement)
+        }
```
Comment (timche) [inline src/content/index.js:177]: Please keep this outside `} else if (pages.isPlayerProfile()) {`.

File: src/content/helpers/player-profile.js (hunk 1 above)
Comment (Morabotti) [reply src/content/helpers/player-profile.js:13]: Lets say you are currently in path `/en/players/username`. Current upstream code returns null with `getPlayerProfileNickname()` method. This method worked fine with `/en/players//stats/csgo`

Comment (Morabotti) [reply src/content/helpers/player-profile.js:13]: Okay so lets say you are in your profile: `/en/players/username`. This current upstream method `getPlayerProfileNickname` does not work with this path and only works when client navigates one of the sub-routes of the profile eg. `/en/players/username/stats/csgo` and since we want to show the badge always on profile, not on sub-routes only, this is required. 

This is 'dirty' batch and could be handled with better regex query. I couldn't figure out a working regex query to this problem.

Comment (timche) [reply src/content/helpers/player-profile.js:13]: The `master` regex detects the nickname for all player profile paths: https://regex101.com/r/UzU8v4/1

I've also tested your branch in the browser and the badge is added on all player profile links.

File: src/content/index.js [hunk 4]
```diff
@@ -161,6 +166,8 @@ function observeBody() {
         })
         addPlayerProfileDownloadDemo(mainContentElement)
         addPlayerProfileExtendedStats(mainContentElement)
+      } else if (pages.isPlayerProfile()) {
+        addPlayerProfileBadge(mainContentElement)
```
Comment (timche) [inline src/content/index.js:170]: I'm sorry, I didn't realize that it was actually correct when it was nested, so `addPlayerProfileBadge` is only called once when it's the player profile in general. Would you mind reverting this back to the previous state please.

File: src/content/index.js [hunk 5]
```diff
@@ -88,6 +90,8 @@ function observeBody() {
         addPlayerProfileDownloadDemo(modalElement)
         addProfileMatchesEloPoints(modalElement)
         addPlayerProfileExtendedStats(modalElement)
+      } else if (modals.isPlayerProfile()) {
+        addPlayerProfileBadge(modalElement)
```
Comment (timche) [inline src/content/index.js:94]: I'm sorry, I didn't realize that it was actually correct when it was nested, so `addPlayerProfileBadge` is only called once when it's the player profile in general. Would you mind reverting this back to the previous state please.

File: src/content/helpers/badge.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+export function getPlayerBadges() {
+  if (!playerBadges) {
+    const vips = store.get('vips')
+
+    playerBadges = [...defaultProfiles, ...vips.map(addVIP)]
+  }
+
+  return playerBadges
+}
```
Comment (timche) [inline src/content/helpers/badge.js:37]: There could be another function `getPlayerBadge(playerId)` that returns `playerBadges.find(({ guid }) => guid === playerId)` to reduce some duplicated logic.

File: src/content/helpers/player-profile.js (hunk 1 above)
Comment (Morabotti) [reply src/content/helpers/player-profile.js:13]: > I've also tested your branch in the browser and the badge is added on all player profile links.

My branch does this minor trick to get it working but see the images:

Path: `/en/players/Morabotti`
matchingProblem

Path: `/en/players/Morabotti/stats/csgo`
matchingProblem2

As you can see, for some reason the first match does return null even though its exactly same regex query as masters. I've working regex query that returns correct name for both of paths, is this good?:

possibleFix
possibleFix2

Comment (timche) [reply src/content/helpers/player-profile.js:13]: Try this regex and the "hack" can be removed. Worked for me.

```diff
diff --git a/src/content/helpers/player-profile.js b/src/content/helpers/player-profile.js
index 2c0f7a0..d26d868 100644
--- a/src/content/helpers/player-profile.js
+++ b/src/content/helpers/player-profile.js
@@ -2,14 +2,7 @@
 import { getCurrentPath } from './location'

 export const getPlayerProfileNickname = path => {
-  const match = /players(?:-modal)?\/([A-Za-z0-9_-]+)/.exec(path || getCurrentPath())
-
-  if (!match || !match[1]) {
-    const profile = /players(?:-modal)?\/([^/]+)$/.exec(
-      path || getCurrentPath()
-    )
-    return profile && profile[1]
-  }
+  const match = /players(?:-modal)?\/([A-Za-z0-9_-]+)(?:\/)?/.exec(path || getCurrentPath())

   return match && match[1]
 }
```

**Comment by Morabotti** [comment]
Should be fine now. Thanks for that regex query.
